azurerm/internal/services/datafactory/data_factory_linked_service_sftp_resource.go

```diff
@@ -184,6 +195,14 @@ func resourceDataFactoryLinkedServiceSFTPCreateUpdate(d *schema.ResourceData, me
 		sftpLinkedService.AdditionalProperties = v.(map[string]interface{})
 	}
 
+	if v, ok := d.GetOk("skip_host_key_validation"); ok {
```

Comment:
Shouldn't we move it to the place after `sftpProperties` is defined, and assign to it - rather than the `sftpLinkdService`. Also, can we be consistent to set the pointer to the `sftpProperties` (though it should work to assign the value as the receiver's type is `interface{}`).

Another point is that the `d.GetOk()` on root level properties always give you `ok = true`, so there is no need to do the check here. If the property is absent, then it gives you its default value for the specific type.

Based on this, we need to further check that the default value of the `SkipHostKeyValidation` is `false`, which is the case as documented in the API spec, so that we do not change the behavior for existing configs. However, due to the case that if `SkipHostKeyValidation` is `false`, users have to set the `HostKeyFingerprint`, then how does the original implementation work - given it doesn't set both?

Reply:
I've tried setting default value to `false`, it generated a diff, although I probably should set it to `Computed` then.
The resource itself didn't work at all, looking at the bugreport in question. I can make `skip_host_key_validation` default to `false` and make `host_key_fingerprint` a Required field, but then it would be a breaking change?

Also I'm not sure I understand the suggestion on moving this to after `sftpProperties` - what's the difference?

Reply:
For the default value, I think current implementation is fine given the original resource is not functional at all.

For the movement of the code, the `HostKeyFingerprint` and the `SkipHostKeyValidation` are direct fields in variable `sftpProperties`, rather than `sftpLinkedService`. Though it makes no difference in this case, whilst I'd prefer to make the relationships more clear.